doc/guides/dts/index.rst

```diff
@@ -1,177 +1,249 @@
+preferred to "device tree".
 
-The device tree is a data structure for dynamically describing hardware
-using a Device Tree Source (DTS) data structure language, instead of
-hard coding every detail of a board's hardware into the operating system.
+.. _Devicetree specification: https://www.devicetree.org/
 
-In Linux, DTS is compiled into a compact Device Tree Blob (DTB) using a Device
-Tree Compiler (DTC), then the hardware-describing DTB is passed to the operating
-system at boot time. This allows the same compiled Linux kernel to support
-different hardware configurations within an architecture family (e.g., ARM,
-x86, PowerPC) and moves a significant part of the hardware description out of
-the kernel binary itself.
+This figure shows how Zephyr uses devicetree:
 
-For larger systems, the flexibility this offers offsets the extra runtime memory
-overhead incurred.  But the primary targets for Zephyr
-applications are small micro-controller systems with limited memory
-resources.  So instead of requiring the additional runtime memory
-to store the DTB blob and the code to parse it, the DTS information
-is used at compile time.
+.. figure:: zephyr_dt_build_flow.png
+   :figclass: align-center
 
-Device tree uses a specific format to describe the device nodes in a system.
-This format is described in the `Device Tree Specification`_.
+   Devicetree build flow
 
-.. _Device Tree Specification: https://github.com/devicetree-org/devicetree-specification/releases
+Zephyr's build system can use a devicetree to generate C language code. This
+code generation uses rules in additional files called *devicetree bindings* to
+control how devicetree data is converted to C definitions. The generated code
+can be included by Zephyr :ref:`device drivers <device_drivers>` and other C
+sources. The C macros generated by this process all begin with ``DT_``.
 
-More device tree information can be found at the `device tree repository`_.
+This process differs significantly from how devicetree is used on Linux. The
+Linux kernel would instead read the entire devicetree data structure in its
+binary form, parsing it at runtime in order to load and initialize device
+drivers. Zephyr does not work this way because the size of the devicetree
+binary and associated handling code would be too large to fit comfortably on
+the relatively constrained devices Zephyr supports.
 
-.. _device tree repository: https://git.kernel.org/pub/scm/utils/dtc/dtc.git
+As the name indicates, a devicetree is a tree. Every node in the tree has a
+*path* identifying its location. Like Unix file system paths, devicetree paths
+are strings separated by slashes (``/``), and the root node's path is a single
+slash: ``/``. Otherwise, each node's path is formed by concatenating the node's
+ancestors' names with the node's own name, separated by slashes, like this:
 
+.. code-block:: none
 
-System build requirements
-*************************
+   /ancestor-1-name/ancestor-2-name/node-name
```

Comment:
Maybe a short generic example could be used here, like

```
/dts-v1/;

/ {
	foo = "string";
	node {
		bar = <3>;
	};
};
```

The explanations could then refer back to it when explaining paths and properties.

Reply:
Good idea. I'll rework this.